packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1733,14 +1778,38 @@ export default function buildKernel(
   }
 
   function changeKernelOptions(options) {
-    assertKnownOptions(options, ['defaultReapInterval', 'snapshotInterval']);
+    assertKnownOptions(options, [
+      'defaultReapInterval',
+      'defaultReapGCKrefs',
+      'snapshotInterval',
+    ]);
     kernelKeeper.startCrank();
     try {
       for (const option of Object.getOwnPropertyNames(options)) {
         const value = options[option];
         switch (option) {
           case 'defaultReapInterval': {
-            kernelKeeper.setDefaultReapInterval(value);
+            if (typeof value === 'number') {
+              assert(value > 0, `defaultReapInterval = ${value}`);
+            } else {
+              assert.equal(value, 'never', `defaultReapInterval = ${value}`);
+            }
```

Comment:
```suggestion
            (typeof value === 'number' && value > 0) ||
              value === 'never' ||
              Fail`defaultReapInterval ${value} must be a positive number or "never"`;
```

Reply:
Sorry, that doesn't scan as well for me.. I've resisted our pattern of `goodthing || Fail` in favor of a more English-like `if (badthing) { Fail }` or better yet `assert(goodthing)`. It's harder when the conditional is more complex like this one. Here are several different options (wrapped to the original's column-offset, which imposes an additional brevity/legibility constraint):

```js
            if (
              (typeof value === 'number' && value <= 0) ||
              value !== 'never'
            ) {
              Fail`defaultReapInterval ${value} must be a positive number or "never"`;
            }
            
            if (
              !((typeof value === 'number' && value > 0) || value === 'never')
            ) {
              Fail`defaultReapInterval ${value} must be a positive number or "never"`;
            }
            
            (typeof value === 'number' && value > 0) ||
              value === 'never' ||
              Fail`defaultReapInterval ${value} must be a positive number or "never"`;
              
            if (typeof value === 'number') {
              assert(value > 0, `defaultReapInterval = ${value}`);
            } else {
              assert.equal(value, 'never', `defaultReapInterval = ${value}`);
            }
            
            assert(
              (typeof value === 'number' && value > 0) || value === 'never',
              `defaultReapInterval ${value} must be a positive number or "never"`,
            );
```

I'm not sure why I find it harder to read the `goodthing || Fail` pattern than everyone else here. I think maybe it's a garden-path thing, where I have to read all the way to the end of the sequence to discover the Fail that inverts the logic.. I start out reading `foo || bar && baz || Fail` as "if foo.. or bar, ok, and this baz thing, then we fail? no, wait, we fail if *not* foo, ok then how does baz affect that?" etc. I've got no problem with `assert(goodthing)`, the outcome I'm looking for is announced up front.

Of all those options, I think I'll go with the last one, the (goodthing) pattern reads well, it comes out one line shorter than the original (although one line longer than your suggestion), and it avoids duplicating the message.